python/cudf/cudf/tests/test_copying.py

```diff
@@ -49,6 +53,33 @@ def test_gather_string_col():
     assert result.data.to_host() == ["a", None, "d"]
 
 
+@pytest.mark.parametrize(
+    "dtype",
+    ["bool", "int8", "int16", "int32", "int64", "float32", "float64", "str"],
+)
+def test_repeat(dtype):
+    arr = np.random.rand(10) * 10
+    repeats = np.random.randint(10, size=10)
+    psr = pd.Series(arr).astype(dtype)
+    gsr = cudf.from_pandas(psr)
+    assert_eq(psr, gsr)
+
+    assert_eq(psr.repeat(repeats).reset_index(drop=True), gsr.repeat(repeats))
```

Comment:
We should be able to handle the index here, how hard would it be to have a `cudf::table` version of repeats so we can pass in the index column(s) as well?

Reply:
Not at all. I'll replace the current API with a table based one.